src/test/regress/sql/rollback_to_savepoint.sql

```diff
@@ -0,0 +1,20 @@
+-- Regression test for this issue:
+-- https://github.com/citusdata/citus/issues/3622
+SET citus.shard_count = 4;
+SET citus.next_shard_id TO 1954000;
+CREATE SCHEMA rollback_to_savepoint;
+SET search_path TO rollback_to_savepoint;
+
+CREATE TABLE t(a int);
+SELECT create_distributed_table('t', 'a');
+
+SET statement_timeout = '5s';
+BEGIN;
+INSERT INTO t VALUES (4);
+SAVEPOINT s1;
```

Comment:
I think here we do insertion that would take longer than the preset statement_timeout (5 seconds) so we rollback to the savepoint s1 right? If so it can be useful to add that as a comment `-- trigger statement timeout`

Also why don't we set `statement_timeout` to something even smaller? Maybe `1s`?

Reply:
I have changed it to 2 seconds now and confirmed that also on CI this timeout triggers the bug here: https://app.circleci.com/pipelines/github/citusdata/citus/8959/workflows/ca8559d1-75fa-4bfe-8a25-5765b6d41eff/jobs/126715

NOTE: Without this PR https://github.com/citusdata/citus/pull/3869, this will actually never fail in CI, since we do not compile with asserts.